common/lib/dependabot/clients/bitbucket.rb

```diff
@@ -19,140 +19,266 @@ class Forbidden < StandardError; end
       def self.for_source(source:, credentials:)
         credential =
           credentials.
-          select { |cred| cred["type"] == "git_source" }.
-          find { |cred| cred["host"] == source.hostname }
+            select { |cred| cred["type"] == "git_source" }.
+            find { |cred| cred["host"] == source.hostname }
 
-        new(credentials: credential)
+        new(source: source, credentials: credential)
       end
 
       ##########
       # Client #
       ##########
 
-      def initialize(credentials:)
+      def initialize(source:, credentials:)
+        @source = source
         @credentials = credentials
         @auth_header = auth_header_for(credentials&.fetch("token", nil))
       end
 
       def fetch_commit(repo, branch)
-        path = "#{repo}/refs/branches/#{branch}"
-        response = get(base_url + path)
-
-        JSON.parse(response.body).fetch("target").fetch("hash")
+        if @source.provider == "bitbucket_server"
```

Comment:
Given that most of these methods include this conditional, I think it makes more sense to extract a separate class for bitbucket server?

Reply:
@jurre I've initially implemented it as a separate class, but later saw a TODO

> TODO: When self-hosted Bitbucket is supported this should use `Bitbucket.for_source`

and decided well if you want to have it in a single class okay